### ISSUE TITLE
Delete and not equals filter don't work for multiple documents

### ISSUE THREAD

**Issue by zhoujoetan** [mikefarah/yq#1377]
**Describe the bug**
Delete such as `del(. | select(.a1=="abc"))` and not equals filter such as del(. | select(.a1!="abc")) does not work for yaml files with multiple documents:

Version of yq: 4.27.5
Operating system: mac/
Installed via: pip3 install

**Input Yaml**
Concise yaml document(s) (as simple as possible to show the bug, please keep it to 10 lines or less)
data1.yml:
```yaml
---
a1: abc
a2: def
---
a1: abd
a2: def
```

**Command**
The command you ran:
```
yq 'del(. | select(.a1=="abc"))' data1.yml
```

**Actual behavior**

```yaml
---
a1: abc
a2: def
---
a1: abd
a2: def
```

**Expected behavior**

```yaml
---
a1: abc
a2: def
```

**Command 2**
The command you ran:
```
yq 'del(. | select(.a1!="abd"))' data1.yml
```

**Actual behavior**

```yaml
---
a1: abc
a2: def
---
a1: abd
a2: def
```

**Expected behavior**

```yaml
---
a1: abc
a2: def
```

**Additional context**

**Comment by mikefarah** [comment]
Thanks for raising, will fix in the next release

**Comment by mikefarah** [comment]
Fixed in 4.28.2
